components/eventing-controller/reconciler/backend/reconciler.go

```diff
@@ -0,0 +1,702 @@
+	desiredSecret.ResourceVersion = currentSecret.ResourceVersion
+	if err := r.Update(ctx, desiredSecret); err != nil {
+		r.Log.Error(err, "Cannot update publisher proxy secret")
+		return nil, err
+	}
+
+	return desiredSecret, nil
+}
+
+func newSecret(name, namespace string) *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+}
+
+func getSecretForPublisher(bebSecret *v1.Secret) (*v1.Secret, error) {
+	secret := newSecret(deployment.PublisherName, deployment.PublisherNamespace)
+
+	secret.Labels = map[string]string{
+		deployment.AppLabelKey: AppLabelValue,
+	}
+
+	if _, ok := bebSecret.Data["messaging"]; !ok {
+		return nil, errors.New("message is missing from BEB secret")
+	}
+	messagingBytes := bebSecret.Data["messaging"]
+
+	if _, ok := bebSecret.Data["namespace"]; !ok {
+		return nil, errors.New("namespace is missing from BEB secret")
+	}
+	namespaceBytes := bebSecret.Data["namespace"]
+
+	var messages []Message
+	err := json.Unmarshal(messagingBytes, &messages)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, m := range messages {
+		if m.Broker.BrokerType == "saprestmgw" {
+			if len(m.OA2.ClientID) == 0 {
+				return nil, errors.New("client ID is missing")
+			}
+			if len(m.OA2.ClientSecret) == 0 {
+				return nil, errors.New("client secret is missing")
+			}
+			if len(m.OA2.TokenEndpoint) == 0 {
+				return nil, errors.New("tokenendpoint is missing")
+			}
+			if len(m.OA2.GrantType) == 0 {
+				return nil, errors.New("granttype is missing")
+			}
+			if len(m.URI) == 0 {
+				return nil, errors.New("publish URL is missing")
+			}
+
+			secret.StringData = getSecretStringData(m.OA2.ClientID, m.OA2.ClientSecret, m.OA2.TokenEndpoint, m.OA2.GrantType, m.URI, string(namespaceBytes))
+			break
+		}
+	}
+
+	return secret, nil
+}
+
+func getSecretStringData(clientID, clientSecret, tokenEndpoint, grantType, publishURL, namespace string) map[string]string {
+	return map[string]string{
+		deployment.PublisherSecretClientIDKey:      clientID,
+		deployment.PublisherSecretClientSecretKey:  clientSecret,
+		deployment.PublisherSecretTokenEndpointKey: fmt.Sprintf(TokenEndpointFormat, tokenEndpoint, grantType),
+		deployment.PublisherSecretEMSURLKey:        fmt.Sprintf("%s%s", publishURL, BEBPublishEndpointForPublisher),
+		PublisherSecretEMSHostKey:                  fmt.Sprintf("%s", publishURL),
+		deployment.PublisherSecretBEBNamespaceKey:  namespace,
+	}
+}
+
+func (r *Reconciler) CreateOrUpdatePublisherProxy(ctx context.Context, backend eventingv1alpha1.BackendType) (*appsv1.Deployment, error) {
+	publisherNamespacedName := types.NamespacedName{
+		Namespace: deployment.PublisherNamespace,
+		Name:      deployment.PublisherName,
+	}
+	currentPublisher := new(appsv1.Deployment)
+	var desiredPublisher *appsv1.Deployment
+
+	switch backend {
+	case eventingv1alpha1.NatsBackendType:
+		desiredPublisher = deployment.NewNATSPublisherDeployment(r.cfg.PublisherConfig)
```

Comment:
Set the ownership for the new "desiredPublisher", which will solve the remaining of "publisher" deployment and "publisher" pod after the helm deletion of eventing:
```
helm install eventing ./eventing -n kyma-system

helm delete eventing -n kyma-system
```

Reply:
Will be solved by: https://github.com/kyma-project/kyma/issues/11342